core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java

```diff
@@ -319,13 +319,14 @@ private JarBuildItem buildUberJar(CurateOutcomeBuildItem curateOutcomeBuildItem,
                     .build();
         }
         final ApplicationManifestConfig manifestConfig = ApplicationManifestConfig.builder()
-                .setApplicationModel(curateOutcomeBuildItem.getApplicationModel())
```

Comment:
To not set the main component that would match the module GAV, since in this case we are building an uber-jar with a classifier, which should be the main component. The rest of the dependencies from the model are added a few lines below.